ExternalModule.php

```diff
@@ -11,31 +11,44 @@
 use User;
 
 /**
- * ExternalModule class for Pain Map.
+ * ExternalModule class for Modify Contact Admin Button.
  */
 class ExternalModule extends AbstractExternalModule {
 
     /**
      * @inheritdoc
      */
     function redcap_every_page_top($project_id) {
-        
+
         if($project_id) {
-            
+
             global $user_firstname, $user_lastname, $user_email, $username, $Proj;
-            
+
+            $length = sizeof($this->getSystemSetting('contact-admin-button-parameters-list-key'));
+            $parameter_names = $this->getSystemSetting('contact-admin-button-parameter-name');
+            $parameter_values = $this->getSystemSetting('contact-admin-button-parameter-value');
             $url = $this->getSystemSetting('contact-admin-button-url-key');
-            
+
             $settings = [
-                    "url" => $url,
                     "user_firstname"=> $user_firstname,
                     "user_lastname" => $user_lastname,
                     "email" => $user_email,
                     "project_id" => $project_id,
                     "username" => $username,
             ];
-            
-            $this->sendVarToJS('contactAdminButtonSettings', $settings);
+
+            $value_pairs = array();
+
+            for($i = 0; $i < $length; $i++){
+                if ($parameter_names[$i] and $parameter_values[$i]){
+                    array_push($value_pairs,
```

Comment:
Why not build the URL here? It would allow you to pass only one variable and reduce the complexity of the JS.